app/domain/authentication/authn_gcp/update_authenticator_input.rb

```diff
@@ -63,10 +63,16 @@ def decoded_credentials
 
       def validate_audience
         if audience_parts.length != 3 ||
-            audience_parts[0] != "conjur" ||
-            audience_parts[1] != account
+            audience_parts[0] != "conjur"
           raise Errors::Authentication::AuthnGcp::InvalidAudience, audience
+        elsif audience_parts[1] != account
```

Comment:
Possible index out of range error.

Reply:
In the 2nd if statement the size is 3 for sure :)